### PR TITLE
Submitting Party - Part 1

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#12579 & /bcgov/entity#12829

*Description of changes:*
*NOTE:* - This is an initial commit of the Submitting Party. There will be a PART 2 to wrap up Validations/Review Confirm Table by Gary. 

- Implements Submitting Party form, Updated Party Look up
- Updates Party lookup to accommodate MHR requirements
- Refactors to the Review Confirm View, pulled out Steps into their own Sub Review Components
- fixed Mhr Search cancel btn

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue [hunk 1]
```diff
@@ -1,18 +1,56 @@
...
+    <section id="mhr-add-submitting-party" class="mt-10">
+      <h2>Submitting Party</h2>
+      <p class="mt-2">
+        Provide the name and contact information for the person or business submitting this registration. You can add
+        the submitting party information manually, or, if the submitting party has a Personal Property Registry party
+        code, you can look up the party code or name.
+      </p>
+
+      <!-- Parties Look Up -->
+      <PartySearch isMhrPartySearch />
```
Comment (dimak1) [inline ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue:12]: `isMhrPartySearch` is not defined here.

**Review by dimak1 — APPROVED** [review]
Looks good! Just one comment, not sure if that would be coming in Part 2.

File: ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue:12]: This is a prop

This is the same as `:isMhrPartySearch="true"`

Comment (cameron-eyds) [reply ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue:12]: Sorry, to add to that, the prop (all props) are defined on their respective components. 

In this case, it's defined with a default `false` value, so we don't have to refactor this components other implementations with this additional prop and so on.

**Review by doug-lovett — APPROVED** [review]
Very nice.

File: ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue (hunk 1 above)
Comment (dimak1) [reply ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue:12]: Yea, I saw the prop, just wanted to check if we need to pass it (as it is always false).

Comment (cameron-eyds) [reply ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue:12]: Ah i see!
Yessir, as mentioned above, the inclusion of the prop here is the same as saying it's true (`:isMhrPartySearch="true"`)

Comment (dimak1) [reply ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue:12]: Ok, I think I got it. You anticipate using `PartySearch` component elsewhere with an option to pass different values for `isMhrPartySearch` prop, and in this case, we want `true`.

Comment (cameron-eyds) [reply ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue:12]: It's currently implemented in other flows, yes. 
We didn't want to impact those other flows but have some unique interactions within `PartySearch` and its subsequent child components for our MHR flow implementation. 

Yessir, absolutely, we want `true` in this case but also want to default too false, so we don't have to include this prop in other implementations.

**Review by dimak1 — APPROVED** [review]

**Pull request merged**